src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs

```diff
@@ -1116,22 +1138,77 @@ void GetPgoOffsetAndVersion(int offset, out int version, out int pgoDataOffset)
             }
         }
 
-        private void CountRuntimeFunctions(bool[] isEntryPoint)
+        private void CountRuntimeFunctions(bool[] isEntryPoint, IDictionary<int, int[]> dHotColdMap, int firstColdRuntimeFunction)
         {
             foreach (ReadyToRunMethod method in Methods)
             {
                 int runtimeFunctionId = method.EntryPointRuntimeFunctionId;
-                if (runtimeFunctionId == -1)
```

Comment:
Is there any real scenario where <code>dHotColdMap</code> would hold an entry indexed by <code>runtimeFunctionId = -1</code>? It seems to me that we're bailing out both in the <code>then</code> and <code>else</code> branch, why cannot we just keep this upfront check?

Reply:
This does look weird and I don't recall why we did this. I will spend some time on it to figure out what is this about.